src/commands/logs/build.mts

```diff
@@ -0,0 +1,92 @@
+import { OptionValues } from 'commander'
+import inquirer from 'inquirer'
+
+import { log, chalk } from '../../utils/command-helpers.mjs'
+import { getWebSocket } from '../../utils/websockets/index.mjs'
+import BaseCommand from '../base-command.mjs'
+
+export function getName({ deploy, userId }: { deploy: any; userId: string }) {
+  let normalisedName = ''
+  const isUserDeploy = deploy.user_id === userId
+
+  switch (deploy.context) {
+    case 'branch-deploy':
+      normalisedName = 'Branch Deploy'
+      break
+    case 'deploy-preview': {
+      // Deploys via the CLI can have the `deploy-preview` context
+      // but no review id because they don't come from a PR.
+      //
+      const id = deploy.review_id
+      normalisedName = id ? `Deploy Preview #${id}` : 'Deploy Preview'
+      break
+    }
+    default:
+      normalisedName = 'Production'
+  }
+
+  if (isUserDeploy) {
+    normalisedName += chalk.yellow('*')
+  }
+
+  return `(${deploy.id.slice(0, 7)}) ${normalisedName}`
+}
+
+const logsBuild = async (options: OptionValues, command: BaseCommand) => {
+  await command.authenticate()
+  const client = command.netlify.api
+  const { site } = command.netlify
+  const { id: siteId } = site
+  const userId = command.netlify.globalConfig.get('userId')
+
+  const deploys = await client.listSiteDeploys({ siteId, state: 'building' })
+
+  if (deploys.length === 0) {
+    log('No active builds')
+    return
+  }
+
+  let [deploy] = deploys
+  if (deploys.length > 1) {
+    const { result } = await inquirer.prompt({
+      name: 'result',
+      type: 'list',
+      message: `Select a deploy\n\n${chalk.yellow('*')} indicates a deploy created by you`,
+      choices: deploys.map((dep: any) => ({
+        name: getName({ deploy: dep, userId }),
+        value: dep.id,
+      })),
+    })
+
+    deploy = deploys.find((dep: any) => dep.id === result)
+  }
+
+  const { id } = deploy
+
+  const ws = getWebSocket(`wss://socketeer.services.netlify.com/build/logs`)
+
+  ws.on('open', function open() {
+    ws.send(JSON.stringify({ deploy_id: id, site_id: siteId, access_token: client.accessToken }))
+  })
+
+  ws.on('message', (data: string) => {
+    const { message, section, type } = JSON.parse(data)
+    log(message)
+
+    if (type === 'report' && section === 'building') {
+      // end of build
+      ws.close()
+    }
+  })
+
+  ws.on('close', () => {
+    log('---')
+  })
+}
+
+export const createLogsBuildCommand = (program: BaseCommand) =>
+  program
+    .command('logs:deploy')
+    .alias('logs:build')
+    .description('(Beta) Stream the logs of active deploys to the console')
```

Comment:
I think it would be worth clarifying that 'active deploys' can mean a production or deploy preview. Given we refer to what's on production as 'published' I think it's easy to read 'active' as potentially being understood as what's 'published'

Reply:
yeah good call out